pdg/src/builder.rs

```diff
@@ -54,10 +55,12 @@ impl EventKindExt for EventKind {
             StoreValue(ptr) => ptr,
             CopyRef => return None, // FIXME
             ToInt(ptr) => ptr,
+            // Question: this has two pointers, is the input one sufficient?
             Realloc { old_ptr, .. } => old_ptr,
             FromInt(lhs) => lhs,
             Alloc { ptr, .. } => ptr,
             AddrOfLocal(lhs, _) => lhs,
+            // Question: same, is the base pointer enough?
             Offset(ptr, _, _) => ptr,
```

Comment:
>is the base pointer enough?

At one point, we planned to distinguish forward offsets (`OFFSET_ADD`) from backward ones (`OFFSET_SUB`).  Offsets that statically only go forward can be implemented on slices with a subslice operation, but offsets that might go backward need a different type that tracks how far back it can go and remain in bounds.